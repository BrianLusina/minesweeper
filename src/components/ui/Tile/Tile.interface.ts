export type TileProps = {
  imageUrl: string;
  thumbnail: string;
  title: string;
  description: string;
};
