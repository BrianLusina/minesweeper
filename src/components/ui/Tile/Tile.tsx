import { FunctionComponent } from 'react';
import { TileProps } from './Tile.interface';

const Tile: FunctionComponent<TileProps> = ({ title, description, imageUrl, thumbnail }) => {
  return (
    <article className="col-6 col-12-xsmall work-item">
      <a href={imageUrl} className="image fit thumb">
        <img src={thumbnail} alt={title} />
      </a>
      <h3>{title}</h3>
      <p>{description}</p>
    </article>
  );
};

export default Tile;
