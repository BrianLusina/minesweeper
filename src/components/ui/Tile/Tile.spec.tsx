import { render, screen } from '@testing-library/react';
import Tile from './Tile';

describe('Tile', () => {
  test('should render', () => {
    const title = 'Some Title';
    const description = 'Some Description';
    const imageUrl = 'https://someimageurl.png';
    const thumbnail = 'https://somethumbnail.png';

    const { container } = render(
      <Tile title={title} description={description} imageUrl={imageUrl} thumbnail={thumbnail} />,
    );

    expect(container).toMatchSnapshot();

    const titleElement = screen.getByText(title);
    const descriptionElement = screen.getByText(description);
    const imageElement = screen.getByAltText(title);

    expect(titleElement).toBeInTheDocument();
    expect(descriptionElement).toBeInTheDocument();
    expect(imageElement).toBeInTheDocument();
  });
});
