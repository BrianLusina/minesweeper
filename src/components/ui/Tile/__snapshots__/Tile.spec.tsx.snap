// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile should render 1`] = `
<div>
  <article
    class="col-6 col-12-xsmall work-item"
  >
    <a
      class="image fit thumb"
      href="https://someimageurl.png"
    >
      <img
        alt="Some Title"
        src="https://somethumbnail.png"
      />
    </a>
    <h3>
      Some Title
    </h3>
    <p>
      Some Description
    </p>
  </article>
</div>
`;
